src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/jobs/InactivateRecallLicencesServiceTest.kt

```diff
@@ -0,0 +1,126 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.jobs
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContext
+import org.springframework.security.core.context.SecurityContextHolder
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.LicenceService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.TestData
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import java.time.LocalDate
+
+class InactivateRecallLicencesServiceTest {
+  private val licenceRepository = mock<LicenceRepository>()
+  private val licenceService = mock<LicenceService>()
+
+  private val service = InactivateRecallLicencesService(
+    licenceRepository,
+    licenceService,
+  )
+
+  @BeforeEach
+  fun reset() {
+    val authentication = mock<Authentication>()
+    val securityContext = mock<SecurityContext>()
+
+    whenever(authentication.name).thenReturn("smills")
+    whenever(securityContext.authentication).thenReturn(authentication)
+    SecurityContextHolder.setContext(securityContext)
+
+    org.mockito.kotlin.reset(
+      licenceRepository,
+      licenceService,
+    )
+  }
+
+  @Test
+  fun `should inactivate active licence with a PRRD of today`() {
```

Comment:
I probably would just have 1 happy path test here - there's no need to test PRRD related to today or variations vs active etc.. - all that logic is in the query that you're stubbing the response to anyway. 

It's potentially a bit misleading as it suggests that the unit under test would treat these differently when really it just inactivates everything its collaborator returns - kind of leaks implementation details of another class and could easily be missed if we change the query